### ISSUE TITLE
There is an issue with running custom script with app.executeCommand expression 

### ISSUE THREAD

**Issue by vkucenko** [inlife/nexrender#580]
I faced an issue with running custom script with app.executeCommand expression.
Example:
`app.executeCommand(app.findMenuCommandId("Convert Audio to Keyframes"));`

so basically I'm trying to recreate "Audio Amplitude" for audio file.

Error I'm getting:
```aerender ERROR An existing connection was forcibly closed by the remote host.
: Unable to receive at line 454
aerender ERROR: After Effects can not render for aerender. Another instance of aerender, or another script, may be running; or, AE may be waiting for response from a modal dialog, or for a render to complete. Try running aerender without the -reuse flag to invoke a separate instance of After Effects

job rendering failed
Error: Couldn't find a result file: C:\Users\User\AppData\Local\Temp\nexrender\tvNx5jTd9IIEsZYNRN6Rh\result.mov
    at ChildProcess.instance.on (C:\ProgramData\nvm\v10.22.0\node_modules\@nexrender\cli\node_modules\@nexrender\core\src\tasks\render.js:174:31)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
    at Socket.stream.socket.on (internal/child_process.js:389:11)
    at Socket.emit (events.js:198:13)
    at Pipe._handle.close (net.js:607:12)
```
the same script works fine if I run it via AF

**Comment by inlife** [comment]
Hey,

Seems like aerender does not like when there is an open GUI window within the after effects while it's being rendered. At least judging by the error message.

**Comment by vkucenko** [comment]
thank you for your comment
do you know a way how to make it works? or maybe another approach of "Audio Amplitude" creation

**Comment by inlife** [comment]
Not sure to be honest. You could try researching this problem for aerender specifically, people on After Effects forums might've already solved it.

**Comment by avsecuri** [comment]
@vkucenko Do you have any solution yet? I have been research for many days with no results

**Comment by vkucenko** [comment]
@avsecuri 
unfortunately I don't

**Comment by eddienubes** [comment]
Hi guys, sorry for brining up a sort of old issue, but I've just experienced the same behaviour trying to execute 
```
app.executeCommand(app.findMenuCommandId("Convert Audio to Keyframes"));
``

Environment:
Windows 10
Adobe After Effects 2023

Tried to replicate the same thing within MacOS and AE 22 and I didn't receive any error but the command execution line was completely ignored thus I didn't see any keyframe layers later on.

Still looking for a possible solution 😞

**Comment by deSenaJr** [comment]
> Hi guys, sorry for brining up a sort of old issue, but I've just experienced the same behaviour trying to execute
> 
> ```
> app.executeCommand(app.findMenuCommandId("Convert Audio to Keyframes"));
> ```
> 
> Environment: Windows 10 Adobe After Effects 2023
> 
> Tried to replicate the same thing within MacOS and AE 22 and I didn't receive any error but the command execution line was completely ignored thus I didn't see any keyframe layers later on.
> 
> Still looking for a possible solution 😞

Hi Eddie, have you had a chance to solve this issue? I am bumping on the same problem!

**Comment by eddienubes** [comment]
[deSenaJr](https://github.com/deSenaJr) Hi!
Unfortunately, there is no way to execute this kind of commands (some of UI buttons don't work at all when called from a script)

For my specific use case of creating a dynamic audio visualisation I analysed audio file manually thus retrieving all the keyframe values by myself and then passed them as an array of values to the script. 

In the script itself I iterated over these approximately 5 thousand values for 3 min audio clip and manually set them as keyframes. 

Basically, each value in this array is the audio's amplitude value at the particular point of time.

There is a code snippet you might find helpful from my repository -> https://github.com/MusyDrop/audio-meta-service/blob/master/src/analyzer/audio-metadata-queue.processor.ts

**Comment by deSenaJr** [comment]
Hey thanks for the guidance!
After some more bug hunting, I found out that my problems derived from
losing the comp active status during the repetitive execution of the
command. Solving it was simple with "comp.openInViewer()" before some calls.
But thanks for the interesting read, learned a ton,.and the smooth() ease()
tips, very helpful!
Have a great week

**Comment by njho** [comment]
@eddienubes! I take it that you haven't changed your approach given the comment above? 

I'm running into the same issue.

Would you be able to share how you used the `compressedRms` data and place the keyframes in the `Right Channel` `Left Channel` and `Both Channels` inside of AE?

![image](https://github.com/inlife/nexrender/assets/16584869/9094ddaf-f547-405b-b2ea-5f1c29d465bb)

**Comment by eddienubes** [comment]
@njho 
Take a look at this line to compose your settings and pass keyframes down to JSX script. https://github.com/MusyDrop/render-worker/blob/9b53799f1855c2c8ff8a11a1e95d3e2f9a3ca704/src/jobs/jobs.controller.ts#L172

Also, the JSX itself is in the assets folder.

**Comment by njho** [comment]
@eddienubes amazing thank you, looking forward to digging into this
